src/controllers/krate/search.rs

```diff
@@ -88,13 +88,15 @@ pub fn search(req: &mut dyn Request) -> CargoResult<Response> {
         );
     }
 
-    if let Some(kw) = params.get("keyword") {
+    if let Some(kws) = params.get_all("keyword") {
+        use diesel::dsl::any;
+        let names: Vec<_> = kws.iter().map(|name| name.to_lowercase()).collect();
         query = query.filter(
             crates::id.eq_any(
                 crates_keywords::table
                     .select(crates_keywords::crate_id)
                     .inner_join(keywords::table)
-                    .filter(::lower(keywords::keyword).eq(::lower(kw))),
+                    .filter(::lower(keywords::keyword).eq(any(names))),
```

Comment:
I actually would have expected searching for multiple keywords to be `any`, not `all` -- which maybe raises the question of what is the general expectation here, if this should have another name

Reply:
I asked the OP and they said they expected it to be the intersect operation, not the union...but for some reason changing `any` to `all` doesn't "just work" like I was hoping it would so I'm still investigating.

I agree that my intuition was that sending multiple keywords to the search endpoint would be an `any` operation, so I'm open to suggestions about where to put this...probably in `src/controllers/keyword.rs`? maybe a new endpoint there?

Reply:
I'm split on this.  I think my initial intuition was `any`, but that doesn't fit with OP's use case which I think is also reasonable.  I guess when I search, I expect results that match all my search terms (although lately Google sometimes seems insistent that I really don't want to include a particular term).

I think its also worth considering how this looks in the URL.  Currently, this uses multiple `keyword` parameters.  Maybe a space separated list under a single `keyword` parameter would be slightly better.  Or a new `keywords` or `all_keywords` parameter.

So I guess calling `all` doesn't work directly here because of the join, since a `crates_keywords` row can't possibly match multiple keywords at the same time.  @sgrif how would this be expressed in Diesel if we want only crates that match all keywords?  I've seen some SQL examples that would involve joining, grouping by crate, and then `HAVING COUNT (DISTINCT crate.keyword_id) = {keyword_count}`.  That seems a bit... complicated, but I can't think of anything simpler.

Reply:
I wouldn't use a having clause for this in PG personally. I'd do something like

```
.filter(
    crate_keywords::table
        .inner_join(keywords::table)
        .filter(crate_keywords::crate_id.eq(crates::id))
        .select(array_agg(keywords::keyword))
        .single_value()
        .contains(names)
)
```

Reply:
So, I know what `array_agg` does in postgres, but there doesn't seem to be a way to do it through diesel. Should I just drop down to `sql_query` for that? Or is `array_agg` in another crate that I just haven't been able to find?

Reply:
`array_agg` is an aggregate function that returns all values it collects as an array. You can declare any SQL function for use with Diesel's query builder by using [`sql_function!`](http://docs.diesel.rs/diesel/macro.sql_function.html)